pkgs/development/tools/build-managers/ninja/setup-hook.sh

```diff
@@ -1,26 +1,22 @@
 ninjaBuildPhase() {
     runHook preBuild
 
-    if [[ -z "$ninjaFlags" && ! ( -e build.ninja ) ]]; then
-        echo "no build.ninja, doing nothing"
-    else
-        local buildCores=1
+    local buildCores=1
 
-        # Parallel building is enabled by default.
-        if [ "${enableParallelBuilding-1}" ]; then
-            buildCores="$NIX_BUILD_CORES"
-        fi
+    # Parallel building is enabled by default.
+    if [ "${enableParallelBuilding-1}" ]; then
+        buildCores="$NIX_BUILD_CORES"
+    fi
 
-        # shellcheck disable=SC2086
-        local flagsArray=( \
-            -j"$buildCores" -l"$NIX_BUILD_CORES" \
-            $ninjaFlags "${ninjaFlagsArray[@]}" \
-            $buildFlags "${buildFlagsArray[@]}")
+    local flagsArray=(
+        -j$buildCores -l$buildCores
```

Comment:
I am hoping it has the same behavior as in make? The only difference is it is autodetected when left out in ninja.

Reply:
There is no change is behaviour, but there is also no need to change `-l$NIX_BUILD_CORES` to `-l$buildCores`. I don't want to see `-l1` in the command line because `-l` should be equal to the total number of busy cores across all concurrent builds; it's value should not be affected by the level of parallelism of a particular build.